zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaaltaakPlugin.kt

```diff
@@ -16,21 +16,254 @@
+class PortaaltaakPlugin(
+    private val objectManagementService: ObjectManagementService,
+    private val pluginService: PluginService,
+    private val valueResolverService: ValueResolverService,
+    private val processDocumentService: ProcessDocumentService,
+    private val zaakInstanceLinkService: ZaakInstanceLinkService,
+    private val taskService: TaskService
+) {
 
     @PluginProperty(key = "notificatiesApiPluginConfiguration", secret = false)
     lateinit var notificatiesApiPluginConfiguration: NotificatiesApiPlugin
 
     @PluginProperty(key = "objectManagementConfigurationId", secret = false)
     lateinit var objectManagementConfigurationId: UUID
+
+    @PluginProperty(key = "uploadedDocumentsHandlerProcess", secret = false)
+    lateinit var uploadedDocumentsHandlerProcess: String
+
+    @PluginAction(
+        key = "create-portaaltaak",
+        title = "Create portal task",
+        description = "Create a task for a portal by storing it in the Objecten-API",
+        activityTypes = [ActivityType.USER_TASK_CREATE]
+    )
+    fun createPortaalTaak(
+        delegateTask: DelegateTask,
+        @PluginActionProperty formType: TaakFormType,
+        @PluginActionProperty formTypeId: String?,
+        @PluginActionProperty formTypeUrl: String?,
+        @PluginActionProperty sendData: List<DataBindingConfig>,
+        @PluginActionProperty receiveData: List<DataBindingConfig>,
+        @PluginActionProperty receiver: TaakReceiver,
+        @PluginActionProperty otherReceiver: OtherTaakReceiver?,
+        @PluginActionProperty kvk: String?,
+        @PluginActionProperty bsn: String?
+    ) {
+        val objectManagement = objectManagementService.getById(objectManagementConfigurationId)
+            ?: throw IllegalStateException("Object management not found for portaal taak")
+
+        val objectenApiPlugin = pluginService.createInstance(
+            PluginConfigurationId
+                .existingId(objectManagement.objectenApiPluginConfigurationId)
+        ) as ObjectenApiPlugin
+
+        val portaalTaak = TaakObject(
+            listOf(getTaakIdentification(delegateTask, receiver, otherReceiver, kvk, bsn)),
+            getTaakData(delegateTask, sendData),
+            delegateTask.name,
+            TaakStatus.OPEN,
+            getTaakForm(formType, formTypeId, formTypeUrl),
+            delegateTask.id
+        )
+
+        val node: JsonNode = jacksonObjectMapper().convertValue(portaalTaak)
+        val x = true
+        //TODO: create actual object
+        //objectenApiPlugin.create
+    }
+
+    @PluginAction(
+        key = "complete-portaaltaak",
+        title = "Complete Portaaltaak",
+        description = "Complete portal task and update status on Objects Api",
+        activityTypes = [ActivityType.SERVICE_TASK_START]
+    )
+    fun completePortaalTaak(delegateExecution: DelegateExecution) {
+        val verwerkerTaakId = delegateExecution.getVariable("verwerkerTaakId") as String
```

Comment:
All of these can be empty, but I think they should always be provided. So if these values are null, we should throw an exception.